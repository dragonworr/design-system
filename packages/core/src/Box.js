import styled from 'styled-components'
import { space, width, textAlign } from 'styled-system'
import { applyVariations, color, deprecatedColorValue } from './utils'

const Box = styled.div`
  ${space} ${width} ${textAlign}
  ${applyVariations('Box')}
  ${color}
`

Box.displayName = 'Box'

Box.propTypes = {
  ...space.propTypes,
  ...width.propTypes,
  color: deprecatedColorValue(),
  bg: deprecatedColorValue(),
  ...textAlign.propTypes
}

export default Box
