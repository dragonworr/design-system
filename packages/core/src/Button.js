import styled, { css } from 'styled-components'
import PropTypes from 'prop-types'
import { width, space } from 'styled-system'
import {
  mapProps,
  deprecatedPropType,
  applyVariant,
  getPaletteColor,
  getTextColorOn
} from './utils'

const size = props => {
  switch (props.size) {
    case 'small':
      return {
        fontSize: `${props.theme.fontSizes[0]}px`,
        padding: '7px 12px'
      }
    case 'medium':
      return {
        fontSize: `${props.theme.fontSizes[1]}px`,
        padding: '9.5px 18px'
      }
    case 'large':
      return {
        fontSize: `${props.theme.fontSizes[2]}px`,
        padding: '12px 22px'
      }
    default:
      return {
        fontSize: `${props.theme.fontSizes[1]}px`,
        padding: '9.5px 18px'
      }
  }
}

const variants = {
  fill: css`
    background-color: ${getPaletteColor('base')};
    color: ${getTextColorOn('base')};

    &:hover {
      background-color: ${getPaletteColor('dark')};
      color: ${getTextColorOn('dark')};
    }
  `,
  outline: css`
    color: ${getPaletteColor('base')};
    box-shadow: inset 0 0 0 2px ${getPaletteColor('base')};
    background-color: transparent;

    &:hover {
      color: ${getPaletteColor('dark')};
      box-shadow: inset 0 0 0 2px ${getPaletteColor('dark')};
      background-color: transparent;
    }
  `,
  disabled: css`
    background-color: ${getPaletteColor('light')};
    color: ${getTextColorOn('light')};
    cursor: default;

    &:hover {
      background-color: null;
    }
  `
}

const Button = mapProps(({ fullWidth, ...props }) => ({
  width: fullWidth ? 1 : undefined,
  ...props
}))(styled.button`
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: ${props => props.theme.radius};
  border-width: 0;
  border-style: solid;

  &:disabled {
    opacity: 0.25;
  }

  ${applyVariant('Button', variants)}

  ${width} ${size} ${space};
`)

Button.propTypes = {
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  ...width.propTypes,
  ...space.propTypes,
  fullWidth: deprecatedPropType('width'),
  variant: PropTypes.oneOf(Object.keys(variants)),
  color: PropTypes.string
}

Button.defaultProps = {
  color: 'primary',
  variant: 'fill'
}

Button.displayName = 'Button'

export default Button
