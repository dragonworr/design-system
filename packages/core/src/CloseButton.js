import React from 'react'
import PropTypes from 'prop-types'
import IconButton from './IconButton'
import { Close } from 'pcln-icons'
import { refPropType } from './utils'

const CloseButton = props => (
  <IconButton {...props} icon={<Close size={props.size} />} />
)

CloseButton.defaultProps = {
  size: 24,
  title: 'close'
}

CloseButton.propTypes = {
  onClick: PropTypes.func,
  size: PropTypes.number,
  dsRef: refPropType,
  title: PropTypes.string
}

CloseButton.displayName = 'CloseButton'

export default CloseButton
