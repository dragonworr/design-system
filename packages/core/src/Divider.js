import styled from 'styled-components'
import { applyVariations, getPaletteColor, deprecatedColorValue } from './utils'
import { space, width, borderColor } from 'styled-system'

const Divider = styled.hr`
  border: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-color: ${({ color, borderColor, ...props }) =>
      getPaletteColor('base')({ color: borderColor || color, ...props })}
    ${space} ${width};
  ${applyVariations('Divider')}
`

Divider.displayName = 'Divider'

Divider.defaultProps = {
  color: 'border',
  ml: 0,
  mr: 0
}

Divider.propTypes = {
  ...space.propTypes,
  ...width.propTypes,
  ...borderColor.propTypes,
  color: deprecatedColorValue()
}

export default Divider
