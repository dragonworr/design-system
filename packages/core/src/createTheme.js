export default theme => {
  return {
    ...theme,
    componentStyles: {
      buttonPrimary: {
        backgroundColor: theme.palette.primary,
        color: theme.colors.white,
        '&:hover': {
          backgroundColor: theme.colors.darkBlue
        },
        ...((theme.componentStyles || {}).buttonPrimary || {})
      },
      buttonSecondary: {
        backgroundColor: theme.palette.secondary,
        color: theme.colors.white,
        '&:hover': {
          backgroundColor: theme.colors.darkGreen
        },
        ...((theme.componentStyles || {}).buttonSecondary || {})
      },
      buttonDefault: {
        backgroundColor: theme.palette.primary,
        color: theme.colors.white,
        '&:hover': {
          backgroundColor: theme.colors.darkBlue
        },
        ...((theme.componentStyles || {}).buttonDefault || {})
      },
      buttonDisabled: {
        backgroundColor: theme.colors.lightGray,
        color: theme.palette.textPrimary,
        cursor: 'default',
        '&:hover': {
          backgroundColor: null
        },
        ...((theme.componentStyles || {}).buttonDisabled || {})
      },
      buttonNegative: {
        backgroundColor: theme.palette.error,
        color: theme.colors.white,
        '&:hover': {
          backgroundColor: theme.colors.darkRed
        },
        ...((theme.componentStyles || {}).buttonNegative || {})
      },
      linkPrimary: {
        color: theme.palette.primary,
        '&:hover': {
          textDecoration: 'underline'
        },
        ...((theme.componentStyles || {}).linkPrimary || {})
      },
      linkSecondary: {
        color: theme.palette.secondary,
        '&:hover': {
          textDecoration: 'underline'
        },
        ...((theme.componentStyles || {}).linkSecondary || {})
      },
      linkTitle: {
        color: theme.colors.red,
        '&:hover': {
          textDecoration: 'underline'
        },
        ...((theme.componentStyles || {}).linkTitle || {})
      },
      linkNavigation: {
        color: theme.palette.textPrimary,
        '&:hover': {
          textDecoration: 'underline'
        },
        ...((theme.componentStyles || {}).linkNavigation || {})
      }
    }
  }
}
