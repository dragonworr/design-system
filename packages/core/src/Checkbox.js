import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import theme from './theme'
import Icon from './Icon'
import Box from './Box'
import { getPaletteColor } from './utils'

const Checkbox = props => {
  const { disabled, size } = props
  return (
    <CheckBoxWrapper color={props.color} disabled={disabled}>
      <StyledInput type="checkbox" {...props} />
      <Icon name="BoxChecked" size={size} data-name="checked" />
      <Icon name="BoxEmpty" size={size} data-name="empty" />
    </CheckBoxWrapper>
  )
}

const CheckBoxWrapper = styled(Box)`
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: ${props =>
    props.disabled
      ? getPaletteColor('border.base')(props)
      : getPaletteColor('border.light')(props)};

  svg[data-name='checked'] {
    display: none;
  }

  > input:checked {
    & ~ svg[data-name='checked'] {
      display: inline-block;
      color: ${props =>
        props.disabled
          ? getPaletteColor('border.base')(props)
          : getPaletteColor('base')(props)};
    }

    & ~ svg[data-name='empty'] {
      display: none;
    }
  }
`

CheckBoxWrapper.defaultProps = {
  theme
}

const StyledInput = styled.input`
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
`

Checkbox.displayName = 'Checkbox'

Checkbox.propTypes = {
  id: PropTypes.string.isRequired,
  size: PropTypes.number,
  onChange: PropTypes.func.isRequired,
  color: PropTypes.string
}

Checkbox.defaultProps = {
  size: 20,
  color: 'primary'
}

export default Checkbox
