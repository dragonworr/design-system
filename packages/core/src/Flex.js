import styled from 'styled-components'
import Box from './Box'
import PropTypes from 'prop-types'
import {
  space,
  width,
  color,
  alignItems,
  justifyContent,
  flexWrap,
  flexDirection
} from 'styled-system'
import theme from './theme'
import { mapProps, deprecatedPropType } from './utils'

const Flex = mapProps(({ wrap, align, justify, ...props }) => ({
  flexWrap: wrap ? 'wrap' : undefined,
  alignItems: align,
  justifyContent: justify,
  ...props
}))(styled(Box)`
  display: flex;
  ${alignItems} ${justifyContent}
  ${flexDirection}
  ${flexWrap}
`)

Flex.propTypes = {
  ...space.propTypes,
  ...width.propTypes,
  // color: PropTypes.string,
  ...alignItems.propTypes,
  ...justifyContent.propTypes,
  ...flexWrap.propTypes,
  ...flexDirection.propTypes,
  // deprecated
  wrap: deprecatedPropType('flexWrap'),
  align: deprecatedPropType('alignItems'),
  justify: deprecatedPropType('justifyContent')
}

Flex.displayName = 'Flex'

export default Flex
