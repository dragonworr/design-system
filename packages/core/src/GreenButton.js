import React from 'react'
import Button from './Button'

const GreenButton = props => <Button variation="secondary" {...props} />

GreenButton.displayName = 'GreenButton'

export default GreenButton
