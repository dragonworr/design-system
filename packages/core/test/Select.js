import React from 'react'
import { Select } from '../src'

describe('Select', () => {
  test('renders', () => {
    const json = rendererCreateWithTheme(<Select />).toJSON()
    expect(json).toMatchSnapshot()
  })
})
