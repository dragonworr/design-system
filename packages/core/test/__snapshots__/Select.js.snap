// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders 1`] = `
.c1 {
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-left: -32px;
  color: #4f6f8f;
}

.c3 {
  pointer-events: none;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: inherit;
  background-color: transparent;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #c0cad5;
  margin: 0px;
  padding-left: 12px;
  padding-right: 32px;
  padding-top: 14px;
  padding-bottom: 14px;
  font-size: 14px;
}

.c2:focus {
  outline: none;
  border-color: #007aff;
  box-shadow: 0 0 0 1px #007aff;
}

.c2::-ms-expand {
  display: none;
}

<div
  className="c0 c1"
  width={1}
>
  <select
    className="c2"
    color="primary"
    fontSize={1}
  />
  <svg
    aria-hidden={true}
    className="c3 c4"
    color="gray"
    fill="currentcolor"
    focusable={false}
    height={24}
    tabIndex={-1}
    viewBox="0 0 24 24"
    width={24}
  >
    <path
      d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
    />
  </svg>
</div>
`;
